cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/RuntimeJobManager.java

```diff
@@ -69,6 +69,15 @@ public interface RuntimeJobManager extends Closeable {
    */
   void kill(ProgramRunInfo programRunInfo) throws Exception;
 
+  /**
+   * Forcefully kills a running job. If the job is already in terminal status, then this method should be a no-op. If
+   * the job does not exist, this method should be a no-op.
+   *
+   * @param runtimeJobDetail runtime Job Detail
+   * @throws Exception thrown if any exception while killing the job
+   */
+  void kill(RuntimeJobDetail runtimeJobDetail) throws Exception;
```

Comment:
Generally not a good idea to have two kill methods that takes different arguments. Can we deprecate the old one? Also, adding new method in an interface requires a default implementation to maintain compatability.

Reply:
Currently what is going on is - 


```

  /**
   * This method ends up actually killing the process because it uses RuntimeJobStatus.RUNNING
   * This method is used in case the intention is to kill the Program, its behavior is different
   * when calling kill(jobDetail) with RuntimeJobStatus.UNKNOWN.
   * */
  @Override
  public void kill(ProgramRunInfo programRunInfo) throws Exception {
    kill(new RuntimeJobDetail(programRunInfo, RuntimeJobStatus.RUNNING));
  }

  /**
   * Unless RuntimeJobStatus is explicitly set as RUNNING, this method gets jobDetail from
   * getDetails() which is equivalent to using RuntimeJobStatus.UNKNOWN
   * */
  @Override
  public void kill(RuntimeJobDetail jobDetail) throws Exception {
    if (jobDetail == null) {
      return;
    }

    RuntimeJobStatus status = jobDetail.getStatus();
    ProgramRunInfo programRunInfo = jobDetail.getRunInfo();

    if (status.isTerminated() || status == RuntimeJobStatus.UNKNOWN) {
      return;
    }

    LOG.debug("Killing program run {} with following configurations: tethered instance name {}, tethered namespace {}.",
              programRunInfo, tetheredInstanceName, tetheredNamespace);
    TetheringControlMessage message = createProgramTerminatePayload(programRunInfo,
                                                                    TetheringControlMessage.Type.KILL_PROGRAM);
    publishToControlChannel(message);
  }

```

Due to this, the  

```
 public void kill(ProgramRunInfo programRunInfo) throws Exception {
```


is the actual kill(), whereas 
```
  public void kill(RuntimeJobDetail jobDetail) throws Exception {
```

is effectively similar to a clean() operation, 
as if we do not pass a jobDetail value, it will call the default getStatus() which returns UNKNOWN, resulting in a return 


```
    if (status.isTerminated() || status == RuntimeJobStatus.UNKNOWN) {
      return;
    }

```

Although I do agree it is good to deprecate the older method, I will try to do away with   

```
public void kill(ProgramRunInfo programRunInfo) throws Exception {
```

I was trying not to break any of the old usages of the old method, but deprecation is less confusing.

Reply:
similar comment here, can't just change the method signature. Instead, deprecate this one and add a default method that calls the new one with running status.

Reply:
still have this comment